_posts/2023-05-09-introducing-timelines-part-1.md

```diff
@@ -0,0 +1,169 @@
+Before a query is performed, each input is mapped to a timeline.
+Every input -- whether events from a stream or table or facts in a table -- can be mapped to a timeline without losing the important temporal information, such as when events happened.
+Events become discrete timelines -- values happening at the time of each event.
+Facts become continuous timelines, reflecting the time during which each fact applied.
+By losslessly representing all kinds of temporal inputs, timelines allow queries to focus on the computation rather than the kind of input.
+
+## Outputting Timelines
+
+After executing a query, the resulting timeline must be output to an external system for consumption.
+<!-- TODO: Link to documentation -->
+The sink for each destination allows configuration of data writing, with specifics depending on the sink and destination (see documentation).
+
+There are several options for converting the timeline into rows of data, affecting the number of rows produced:
+
+1. Include the entire history of changes within a range or just a snapshot of the final value.
+2. Include only events (changes) occurring after a certain time in the output.
+3. Include only events (changes) up to a specific time in the output.
+
+A full history of changes helps visualize or identify patterns in user values over time.
+In contrast, a snapshot at a specific time is useful for online dashboards or classifying similar users.
+
+Including events after a certain time reduces output size when the destination already has data up to that time or when earlier points are irrelevant.
+This is particularly useful when re-running a query to materialize to a data store.
+
+Including events up to a specific time also limits output size and enables choosing a point-in-time snapshot.
+With incremental execution, selecting a time slightly earlier than the current time reduces late data processing.
+
+Both "changed since" and "up-to" options are especially useful with incremental execution, which we will discuss in part 4.
+
+### History
+
+The _history_ -- the set of all points in the timeline -- is useful when you care about past points.
+For instance, this may be necessary to visualize or identify patterns in how the values for each user change over time.
+
+![Diagram showing the conversion of all points in a timeline into a sequence of change events.][timeline_history]
+
+Any timeline may be output as a history.
+For a discrete timeline, the history is the collection of events in the timeline.
+For a continuous timeline, the history contains the points at which a value changes -- it is effectively a change log
+
+### Snapshots
+
+A _snapshot_ -- the value for each entity at a specific point in time -- is useful when you just care about the latest values.
+For instance, when updating a dashboard or populating a feature store.
+
+![Diagram showing the conversion of values at a point in time in a timeline to a snapshot, including interpolation.][timeline_snapshot]
```

Comment:
Text leaks off the white background a bit, but I think you've noted that already

Reply:
It's currently a non issue since the background of the page is white.